exams/management/commands/import_edx_exam_grades.py

```diff
@@ -21,68 +21,61 @@ class Command(BaseCommand):
-                course=course,
-                date_first_schedulable__lte=now,
-                date_last_schedulable__gte=now,
-            )
-        except Course.DoesNotExist:
-            raise CommandError(
-                'There are no eligible exam runs for course "{}"'.format(course_number)
-            )
-        except Course.MultipleObjectsReturned:
-            raise CommandError(
-                'There are multiple eligible exam runs for course "{}"'.format(course_number)
-            )
-
         grade_count = 0
         existing_grades = 0
         for row in reader:
-            user = User.objects.get(username=row['Username'])
-            exam_authorization = ExamAuthorization.objects.get(user=user, exam_run=exam_run)
-            passed = float(row['Grade']) >= exam_run.passing_score
-            defaults = {
-                'passing_score': exam_run.passing_score,
-                'score': float(row['Grade'])*100,
-                'grade': EXAM_GRADE_PASS if passed else EXAM_GRADE_FAIL,
-                'percentage_grade': float(row['Grade']),
-                'passed': passed,
-                'row_data': row,
-                'exam_date': now_in_utc()
-            }
-            _, created = ProctoredExamGrade.objects.update_or_create(
-                user=user,
+            user = User.objects.get(username=row['username'])
+            course_id = row['course_id']
+
+            try:
+                course = Course.objects.get(id=course_id)
+            except Course.DoesNotExist:
+                raise CommandError(
+                    'Could not find a course with number "{}"'.format(course_id)
+                )
+            # should pick the latest past exam run
+            now = now_in_utc()
+            exam_run = ExamRun.objects.filter(
                 course=course,
-                exam_run=exam_run,
-                defaults=defaults
-            )
-            if created:
-                grade_count += 1
+                date_first_schedulable__lte=now
+            ).order_by('-date_last_schedulable').first()
+            if exam_run is None:
+                raise CommandError(
+                    'There are no eligible exam runs for course "{}"'.format(course.title)
+                )
+            exam_authorization = ExamAuthorization.objects.get(user=user, exam_run=exam_run)
+
+            if int(row['no_show']):
                 exam_authorization.exam_taken = True
+                exam_authorization.exam_no_show = True
                 exam_authorization.save()
             else:
-                existing_grades += 1
+                defaults = {
+                    'passing_score': exam_run.passing_score,
+                    'score': float(row['score']),
+                    'grade': row['grade'],
+                    'percentage_grade': float(row['score']) / 100.0 if row['score'] else 0,
+                    'passed': row['grade'].lower() == EXAM_GRADE_PASS,
```

Comment:
Just double checking, is this correct? The previous code calculated this value by comparing the grade to `exam_run.passing_score`

Reply:
Yes, before the it was using the passing score to decide whether the grade is passing. But Peter said that we could just get that value from the spreadsheet, I think Maya calculates this value when adjusting the grades.